R/derive_merged.R

```diff
@@ -0,0 +1,685 @@
+
+  add_data <- filter_if(dataset_add, filter_add) %>%
+    mutate(!!new_var := cat_fun(!!source_var))
+  derive_vars_merged(
+    dataset,
+    dataset_add = add_data,
+    by_vars = by_vars,
+    order = order,
+    new_vars = vars(!!new_var),
+    match_flag = temp_match_flag,
+    mode = mode
+  ) %>%
+    mutate(!!new_var := if_else(temp_match_flag, !!new_var, missing_value, missing_value)) %>%
+    select(-temp_match_flag)
+}
+
+#' Merge an Existence Flag
+#'
+#' Adds a flag variable to the input dataset which indicates if there exists at
+#' least one observation in another dataset fulfilling a certain condition.
+#'
+#' @param dataset_add Additional dataset
+#'
+#'   The variables specified by the `by_vars` parameter are expected.
+#'
+#' @param by_vars Grouping variables
+#'
+#'   *Permitted Values*: list of variables
+#'
+#' @param new_var New variable
+#'
+#'   The specified variable is added to the input dataset.
+#'
+#' @param condition Condition
+#'
+#' @param true_value True value
+#'
+#' @param false_value False value
+#'
+#' @param filter_add Filter for additional data
+#'
+#'   Only observations fulfilling the specified condition are taken into account
+#'   for flagging. If the parameter is not specified, all observations are
+#'   considered.
+#'
+#'   *Permitted Values*: a condition
+#'
+#' @inheritParams derive_vars_merged
+#'
+#' @details
+#'
+#'   1. The additional dataset is restricted to the observations matching the
+#'   `filter_add` condition.
+#'
+#'   1. The new variable is added to the input dataset and set to the true value
+#'   if for the by group at least one observation exists in the (restricted)
+#'   additional dataset where the condition evaluates to `TRUE`. Otherwise, it
+#'   is set to the false value.
+#'
+#' @author Stefan Bundfuss
+#'
+#' @keywords derivation adam
+#'
+#' @export
+#'
+#' @examples
+#'
+#' library(admiral.test)
+#' library(dplyr, warn.conflicts = FALSE)
+#' data("dm")
+#' data("ae")
+#' derive_var_merged_exist_flag(
+#'   dm,
+#'   dataset_add = ae,
+#'   by_vars = vars(STUDYID, USUBJID),
+#'   new_var = AERELFL,
+#'   condition = AEREL == "PROBABLE"
+#' ) %>%
+#' select(STUDYID, USUBJID, AGE, AGEU, AERELFL)
+derive_var_merged_exist_flag <- function(
+  dataset,
+  dataset_add,
+  by_vars,
+  new_var,
+  condition,
+  true_value = "Y",
+  false_value = NA_character_,
+  filter_add = NULL
+) {
+  new_var <- assert_symbol(enquo(new_var))
+  condition <- assert_filter_cond(enquo(condition))
+  filter_add <-
+    assert_filter_cond(enquo(filter_add), optional = TRUE)
+
+  add_data <- filter_if(dataset_add, filter_add) %>%
+    mutate(!!new_var := if_else(!!condition, 1, 0, 0))
+
+  derive_vars_merged(
+    dataset,
+    dataset_add = add_data,
+    by_vars = by_vars,
+    new_vars = vars(!!new_var),
+    order = vars(!!new_var),
+    check_type = "none",
+    mode = "last"
+  ) %>%
+    mutate(!!new_var := if_else(!!new_var == 1, true_value, false_value, false_value))
```

Comment:
false_value is given twice

Reply:
This is intended the first one is for condition is false and the second one for condition is `NA`.

However, I have replaced the second one by `missing_value`.